src/config/backends/extlinux-file.ts

```diff
@@ -0,0 +1,86 @@
+import * as _ from 'lodash';
+
+import { ConfigOptions } from './backend';
+import { ExtLinuxParseError } from '../../lib/errors';
+
+export interface ExtlinuxFile {
+	labels: Label;
+	globals: Directive;
+}
+
+export interface Label {
+	[labelName: string]: Directive;
+}
+
+export interface Directive {
+	[directive: string]: string;
+}
+
+export abstract class ConfigurableDirective {
```

Comment:
Given that we were talking about the ease of creating new directives, could you add a couple of comments here detailing that this is the base class of the directives, to make a new one, implement this etc etc.

Basically just an overview of what its for for any future contributors.

Reply:
I've added documentation to the class abstraction and each directive class.